modules/core/shared/src/main/scala/gem/math/ProperMotion.scala

```diff
@@ -0,0 +1,182 @@
+  * trying to understand the implementation.
+ * @see The pretty good [[https://en.wikipedia.org/wiki/Proper_motion wikipedia]] article
+ * @see Astronomical Almanac 1984 [[https://babel.hathitrust.org/cgi/pt?id=uc1.b3754036;view=1up;seq=141 p.B39]]
+ * @see Astronomy and Astrophysics 134 (1984) [[http://articles.adsabs.harvard.edu/cgi-bin/nph-iarticle_query?bibcode=1984A%26A...134....1L&db_key=AST&page_ind=0&data_type=GIF&type=SCREEN_VIEW&classic=YES p.1-6]]
+ * @param baseCoordinates observed coordinates at `epoch`
+ * @param epoch           time of the base observation; typically `Epoch.J2000`
+ * @param properVelocity  proper velocity '''per year''' in [[RightAscension]] and [[Declination]], if any
+ * @param radialVelocity  radial velocity (km/y, positive if receding), if any
+ * @param parallax        parallax, if any
+ */
+final case class ProperMotion(
+  baseCoordinates: Coordinates,
+  epoch:           Epoch,
+  properVelocity:  Option[Offset],
+  radialVelocity:  Option[Double], // for now
+  parallax:        Option[Angle]
+) {
+
+  /** Coordinates `elapsedYears` fractional epoch-years after `epoch`. */
+  def plusYears(elapsedYears: Double): Coordinates =
+    ProperMotion.properMotion(
+      baseCoordinates,
+      epoch,
+      properVelocity.orEmpty,
+      radialVelocity.orEmpty,
+      parallax.orEmpty,
+      dt
+    )
+
+}
+
+
+object ProperMotion {
+
+  /**
+   * Proper motion correction in model units.
+   * @param baseCoordinates base coordinates
+   * @param epoch           the epoch
+   * @param properVelocity  proper velocity per epoch year
+   * @param radialVelocity  radial velocity (km/sec, positive if receding)
+   * @param parallax        parallax
+   * @param elapsedYears    elapsed time in epoch years
+   * @return Coordinates corrected for proper motion
+   */
+  def properMotion(
+    baseCoordinates: Coordinates,
+    epoch:           Epoch,
+    properVelocity:  Offset,
+    radialVelocity:  Double,
+    parallax:        Angle,
+    elapsedYears:    Double
+  ): Coordinates = {
+    val (ra, dec) = properMotionʹ(
+      baseCoordinates.toRadians,
+      epoch.scheme.lengthOfYear,
+      properVelocity.toRadians,
+      radialVelocity,
+      parallax.toMicroarcseconds.toDouble / 1000000.0,
+      elapsedYears
+    )
+    Coordinates.unsafeFromRadians(ra, dec)
+  }
+
+  // Some constants we need
+  private val secsPerDay  = 86400.0
+  private val auPerKm     = 1.0 / 149597870.0
+  private val radsPerAsec = Angle.fromArcseconds(1).toDoubleRadians
+  private val twoPi       = 6.283185307179586476925286766559
+
+  // We need to do things with little vectors of doubles
+  private type Vec2 = (Double, Double)
+  private type Vec3 = (Double, Double, Double)
+
+  // |+| gives us addition for VecN, but we also need scalar multiplication
+  private implicit class Vec3Ops(a: Vec3) {
+    def *(d: Double): Vec3 =
+      (a._1 * d, a._2 * d, a._3 * d)
+  }
+
+  /**
+   * Proper motion correction in base units.
```

Comment:
This is private ... once we drop down to `Double` it probably makes sense to hide our cards. We'll see if it turns out to be a pain.